doc/modules/ensemble.rst

```diff
@@ -102,24 +102,72 @@ controls the number of iterations of the boosting process::
   >>> clf.score(X_test, y_test)
   0.8965
 
-Available losses for regression are 'squared_error',
-'absolute_error', which is less sensitive to outliers, and
-'poisson', which is well suited to model counts and frequencies. For
-classification, 'log_loss' is the only option. For binary classification it uses the
-binary log loss, also known as binomial deviance or binary cross-entropy. For
-`n_classes >= 3`, it uses the multi-class log loss function, with multinomial deviance
-and categorical cross-entropy as alternative names. The appropriate loss version is
-selected based on :term:`y` passed to :term:`fit`.
+Available losses for regression are 'squared_error', 'absolute_error', which is
+less sensitive to outliers, and 'poisson', which is well suited to model counts
+and frequencies. For classification, 'log_loss' is the only option. For binary
+classification it uses the binary log loss, also known as binomial deviance or
+binary cross-entropy. For `n_classes >= 3`, it uses the multi-class log loss
+function, with multinomial deviance and categorical cross-entropy as alternative
+names. The appropriate loss version is selected based on :term:`y` passed to
+:term:`fit`.
 
 The size of the trees can be controlled through the ``max_leaf_nodes``,
 ``max_depth``, and ``min_samples_leaf`` parameters.
 
 The number of bins used to bin the data is controlled with the ``max_bins``
-parameter. Using less bins acts as a form of regularization. It is
-generally recommended to use as many bins as possible (256), which is the default.
+parameter. Using less bins acts as a form of regularization. It is generally
+recommended to use as many bins as possible (256), which is the default.
+
+The ``l2_regularization`` parameter acts as a regularizer for the loss function,
+and corresponds to :math:`\lambda` in the following regularized objective (see
+equation (2) of [XGBoost]_):
+
+.. math::
+
+      \mathcal{L}(\phi) =  \sum_i l(\hat{y}_i, y_i) + \sum_k \Omega(f_k) \\
+      \text{where} ~ \Omega(f_k) = \gamma T_k  + \frac12 \lambda ||w_k||^2
+
+Here :math:`l` is the loss function; :math:`T_k` is the number of leaves in the
+tree; :math:`f_k` corresponds to the k-th tree in the ensemble of trees (such
+that :math:`\hat{y}_i=\phi(x_i)=\sum_k f_k(x_i)`); and :math:`w_k` is a vector
+of length :math:`T_k` containing the leaf weights.
+
+Notice that :math:`\gamma` penalizes the number of leaves (which makes it a
+smooth version of `max_leaf_nodes` and is not implemented in scikit-learn),
+whereas :math:`\lambda` penalizes the magnitude of the individual tree
+predictions.
+
+**Note**: In practice we don't recommend tuning `l2_regularization`, as it does
```

Comment:
Interesting, I was under the impression it had very little effect. Maybe we can just remove this sentence. Alternatively, we could say that this hyperparameters if less important to tune than others such as `learning_rate` or `max_leaf_nodes`.

Reply:
No, I would just not comment it at all. I'm fine with commenting that learning rate and max leaf nodes are very important parameters.
Better draw attention to positive statements like "this is important", instead of to less important negative ones like "this might not be that important after all".

Reply:
> Alternatively, we could say that this hyperparameters if less important to tune than others such as `learning_rate` or `max_leaf_nodes`.

At the beginning of [1.11.1.2.](https://scikit-learn.org/dev/modules/ensemble.html#gradientboostingclassifier-and-gradientboostingregressor) we say

> The 2 most important parameters of these estimators are `n_estimators` and `learning_rate`.

(Which is also sub-optimal wording as the user may consider tuning `n_estimators`, but that's yet another story). Maybe it is better to completely remove the note proposed in this PR.

Reply:
I think we should rephrase the usage section of HistGradientBoostingC/R from:

> Most of the parameters are unchanged from GradientBoostingClassifier and GradientBoostingRegressor. One exception is the max_iter parameter that replaces n_estimators, and controls the number of iterations of the boosting process.

to something like:

> HistGradientBoostingClassifier and HistGradientBoostingRegressor have some important parameters such as `max_iter` that controls the number of boosting rounds and therefore linearly controls the number of trees in the ensemble, `learning_rate` that controls the incremental impact of each round on the prediction and `max_leaf_nodes` (or other parameters such as `max_depth` and `min_samples_leaf` that all control the size of the trees). While `max_iter` can be left to an arbitrarily large value when enabling early stopping, `learning_rate` and `max_leaf_nodes` typically benefit from being tuned by cross-validated parameter search.

Reply:
I would rather leave this change to another PR, where I can address [@lorentzenchr 's comment](https://github.com/scikit-learn/scikit-learn/pull/28652/files#r1537489269) and possibly refactor the discussion on GBDT and HGBT, as proposed in https://github.com/scikit-learn/scikit-learn/pull/28652#pullrequestreview-1957577755 (if I understood correctly).